spox: init at 0.10.2

cbourjau opened the pull request:
Release note of this release:
https://github.com/Quantco/spox/releases/tag/0.10.2

Spox is a pythonic framework for building ONNX graphs.

## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

cbourjau commented:
Result of `nixpkgs-review pr 289864` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>spox</li>
    <li>spox.dist</li>
  </ul>
</details>

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/3511

AndersonTorres commented:
I have tried to review this but it does not generate any runnable binaries.

AndersonTorres commented:
AFAICR, if this is a Python library then it should be put on python-packages.nix.

SuperSandro2000 commented on pkgs/by-name/sp/spox/package.nix:
```diff
@@ -0,0 +1,37 @@
+  ];
+
+  nativeCheckInputs = with python3Packages; [
+    pytest
```
```suggestion
    pytestCheckHook
```

cbourjau commented:
Thanks for the feedback! If I understood it correctly, packages referenced in `python-packages` are supposed to be defined in `pkgs/development/python-modules`, are they not?

cbourjau committed:
Spox: init at 0.10.2

Release note of this release:
https://github.com/Quantco/spox/releases/tag/0.10.2

Spox is a pythonic framework for building ONNX graphs.

cbourjau commented:
Result of `nixpkgs-review pr 289864` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python311Packages.spox</li>
    <li>python311Packages.spox.dist</li>
  </ul>
</details>

SuperSandro2000 commented on pkgs/development/python-modules/spox/default.nix:
```diff
@@ -0,0 +1,39 @@
+    hash = "sha256-VfxyIRN8nNNEda7zPPRbYKfC7fXV6K45eYsT6vngmWk=";
+  };
+
+  nativeBuildInputs = with python3Packages; [
+    setuptools-scm
+  ];
+
+  propagatedBuildInputs = with python3Packages; [
+    numpy
+    onnx
+    packaging
+  ];
+
+  nativeCheckInputs = with python3Packages; [
+    pytestCheckHook
+    onnxruntime
+  ];
```
```suggestion
  nativeBuildInputs = [
    setuptools-scm
  ];

  propagatedBuildInputs = [
    numpy
    onnx
    packaging
  ];

  nativeCheckInputs = [
    pytestCheckHook
    onnxruntime
  ];
```

SuperSandro2000 commented on pkgs/development/python-modules/spox/default.nix:
```diff
@@ -0,0 +1,39 @@
+{ lib
+, buildPythonPackage
+, python3Packages
+, fetchPypi
+ }:
```
```suggestion
{ lib
, buildPythonPackage
, fetchPypi
, numpy
, onnx
, onnxruntime
, packaging
, pytestCheckHook
, setuptools-scm
```

we cannot use python3Packages here because that draws in python 3.11
